lib/localStorage.ts

```diff
@@ -0,0 +1,87 @@
+// Collection of helper functions to interact with localStorage.
+// See original code in Frontend:
+// https://github.com/guardian/frontend/blob/master/static/src/javascripts/lib/storage.js
+const checkAvailability = (): boolean => {
+  const key = 'local-storage-module-test';
+  try {
+    localStorage.setItem(key, 'graun');
+    localStorage.removeItem(key);
+    return true;
+  } catch (e) {
+    return false;
+  }
+};
+
+const storage = window.localStorage;
+const isAvailable = checkAvailability();
+
+export const get = (key: string) => {
+  if (!isAvailable) {
+    return;
+  }
+
+  let data;
+  // Try and parse the data
+  try {
+    const value = getRaw(key);
+
+    if (value === null || value === undefined) {
+      return null;
+    }
+
+    data = JSON.parse(value);
+
+    if (data === null) {
+      return null;
+    }
+  } catch (e) {
+    remove(key);
+    return null;
+  }
+
+  // Check if the data has an expiration date and remove it if expired
+  if (data.expires && new Date() > new Date(data.expires)) {
+    remove(key);
+    return null;
+  }
+
+  // If data contains a value property, assume that's what we care about
+  if (typeof data.value !== 'undefined') {
```

Comment:
Interesting, I wondered if `!== undefined` was now adequate. And it turns out it *almost* is (as JS now prevents overriding window.undefined) but for certain older browsers errors are still possible unfortunately.